docs/src/DeveloperDocumentation/experimental.md

```diff
@@ -0,0 +1,25 @@
+# Implementing features from scratch
+
+For new features whose interface naturally will change a lot during the
+implementation phase, Oscar.jl has a staging area, the `experimental` folder.
+
+1. The new feature is implemented in the `experimental` folder.
+2. Additionally to the original authors, a maintainer is assigned to guide the
```

Comment:
I just wanted to write: We need a second person only for newbies.

But maybe that's not true.  Maybe even OSCAR veterans could occasionally discuss why some of their code is sitting in experimental for a quite a while?

The main problem with the current version of `experimental` is that the user cannot really distinguish from the actual core.

Reply:
Indeed. Some options come to mind:
- maybe accessing code in experimental should be "guarded" and require `using Oscar.Experimental`, or require writing `Experimantal.FOO` (so `Experimental.galois_group(...)`
  - pro: obviously experimental
  - con: less discoverable
- or maybe the primary functions should have a `@warn_experimental()` call in them, which prints a warning about them being experimental (possibly only the first time they are called)
- ... other ideas?

Reply:
For the discoverability: I still think, code should not spend too much time in `experimental`, then this would matter less. I don't know whether we want to provide a concrete time frame though.

Reply:
Code in experimental will be buggy because it is less tested and no battle proof yet. So the naive user should not use it at all. Certainly not to prove a theorem.